helpers.lua

```diff
@@ -276,7 +276,7 @@ function validate_or_clear_basic_auth_header_provided_by_client()
 
     -- Try to authenticate the user,
     -- or remove the Auth header if not valid
-    _, _, user, password = string.find(ngx.decode_base64(b64_cred), "^(.+):(.+)$")
+    _, _, user, password = string.find(ngx.decode_base64(b64_cred), "^([^:].-):(.+)$")
```

Comment:
Ugh I've read the discussion in diagonal but not sure to understand the whole thing about `([^:].-)` ... not sure to understand why this works ... Why not using `([^:]+)` o_O ?

Reply:
Correct. `([^:]+)` is equivalent and makes sense... I was blinded by that `.-` approach.

Reply:
It is completely different look at it, since it means "find as many non-semicolon characters as possible, at least one"but it gets the job done in the same way as `[^:].-`.
Keep in mind that one of major questions here is: Is "" user allowed? This approach forbids it.

Reply:
hmyeah but `+` also means "at least one char" so it won't match empty string ;P

Reply:
None of these would match empty string.
Some of those could match empty username (ewilly's first one and `(.-)`) but it was not permited before anyway.

Both `([^:]+)` and `([^:.-)` gets the job done and I am ok with these. `([^:]+)` makes a bit more sense. I don't expect "" user to be something useful anyway, I just wanted to point it out so I got somebody thinking about it because honestly I don't know the answer.

Reply:
commit updated with @alexAubin proposal which is working for me too.